idseq_pipeline/commands/non_host_alignment_functions.py

```diff
@@ -108,6 +113,78 @@
 TAX_LEVEL_GENUS = 2
 
 # convenience functions
+def count_lines_in_paired_files(input_files):
+    known_nlines = None
+    for input_file in input_files:
+        nlines = int(execute_command_with_output("wc -l %s" % input_file).strip().split()[0])
+        if known_nlines != None:
+            assert nlines == known_nlines, "Mismatched line counts in supposedly paired files: {}".format(input_files)
+        known_nlines = nlines
+    return nlines
+
+def subsample_single_fasta(input_file, records_to_keep, output_file):
+    record_number = 0
+    kept_read_ids = []
+    input = open(input_file, 'rb')
+    output = open(output_file, 'wb')
+    sequence_name = input.readline()
+    sequence_data = input.readline()
+    while len(sequence_name) > 0 and len(sequence_data) > 0:
+       if record_number in records_to_keep:
+            output.write(sequence_name)
+            output.write(sequence_data)
+            kept_read_ids += sequence_name.rstrip()
+       sequence_name = input_fasta_f.readline()
+       sequence_data = input_fasta_f.readline()
+       record_number += 1
+    input.close()
+    output.close()
+    return kept_read_ids
+
+def extract_read_ids_from_fasta(input_file, read_ids_to_keep, output_file):
+    input = open(input_file, 'rb')
+    output = open(output_file, 'wb')
+    sequence_name = input.readline()
+    sequence_data = input.readline()
+    while len(sequence_name) > 0 and len(sequence_data) > 0:
+       sequence_basename = sequence_name.rstrip().rsplit('/', 1)[0]
+       if sequence_basename in read_ids_to_keep:
+            output.write(sequence_name)
+            output.write(sequence_data)
+            kept_read_ids += sequence_basename
+       sequence_name = input_fasta_f.readline()
+       sequence_data = input_fasta_f.readline()
+    input.close()
+    output.close()
+    assert set(kept_read_ids) == set(read_ids_to_keep), "Not all desired read IDs were found in the file: {}".format(input_file)
```

Comment:
This function is almost identical to the previous one.   Could be nice to unify them.   Also, in python it's better to use "with" statement with files, handles resource management cleanly regardless of exceptions etc.